control_toolbox/include/control_toolbox/low_pass_filter.hpp

```diff
@@ -186,13 +200,57 @@ inline bool LowPassFilter<geometry_msgs::msg::WrenchStamped>::update(
   return true;
 }
 
+template <>
+inline bool LowPassFilter<std::vector<double>>::update(
+  const std::vector<double> & data_in, std::vector<double> & data_out)
+{
+  if (!configured_)
+  {
+    throw std::runtime_error("Filter is not configured");
+  }
+  // If this is the first call to update initialize the filter at the current state
+  // so that we dont apply an impulse to the data.
+  // This also sets the size of the member variables to match the input data.
+  if (filtered_value.empty())
+  {
+    filtered_value = data_in;
+    filtered_old_value = data_in;
+    old_value = data_in;
+  }
+  else
+  {
+    assert(
+      data_in.size() == filtered_value.size() &&
+      "Internal data and the data_in doesn't hold the same size");
+    assert(data_out.size() == data_in.size() && "data_in and data_out doesn't hold same size");
+  }
+
+  // Filter each value in the vector
+  for (std::size_t i = 0; i < data_in.size(); i++)
+  {
+    data_out[i] = b1_ * old_value[i] + a1_ * filtered_old_value[i];
+    filtered_old_value[i] = data_out[i];
+    old_value[i] = data_in[i];
```

Comment:
```suggestion
    if(!std::finite(data_in[i]))
    {
      old_value[i] = data_in[i];
    }
```
What do you think about protecting the filter from any NaNs?